### PR TITLE
[Feat]merge ps-cs, ps-data-source-manager and ps-metadataquery to ps-publicservice

### PR DESCRIPTION
**Pull request by jackxu2011**
### What is the purpose of the change

Reduce service

### Related issues/PRs

Related issues: #2928 

### Brief change log

- merge ps-cs, ps-data-source-manager and ps-metadataquery to ps-publicservice
- merge config file to ps-publicservice

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-dist/package/sbin/linkis-stop-all.sh [hunk 1]
```diff
@@ -63,21 +63,21 @@ SERVER_IP=$GATEWAY_INSTALL_IP
...
+#  #linkis-ps-data-source-manager
+#  SERVER_NAME="ps-data-source-manager"
+#  SERVER_IP=$DATASOURCE_MANAGER_INSTALL_IP
+#  stopApp
+#
+#  #linkis-ps-metadataquery
+#  SERVER_NAME="ps-metadataquery"
+#  SERVER_IP=$METADATA_QUERY_INSTALL_IP
+#  stopApp
+#fi
```
Comment (casionone) [inline linkis-dist/package/sbin/linkis-stop-all.sh:80]: remove it directly?

**Comment by casionone** [comment]
Follow-up to do:
Official website documentation needs to be updated
The linkis-dist/helm/charts/linkis helm configuration file also needs to be modified

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ object RPCConfiguration {
 
   val PUBLIC_SERVICE_LIST: Array[String] = CommonVars(
     "wds.linkis.gateway.conf.publicservice.list",
-    "query,jobhistory,application,configuration,filesystem,udf,variable,microservice,errorcode,bml,datasource"
+    "cs,data-source-manager,metadataquery,query,jobhistory,application,configuration,filesystem,udf,variable,microservice,errorcode,bml,datasource"
```
Comment (peacewong) [inline linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala:66]: cs should contextservice

File: linkis-dist/package/conf/linkis-ps-publicservice.properties [hunk 3]
```diff
@@ -13,25 +13,35 @@
...
+
 ##workspace
 wds.linkis.workspace.filesystem.hdfsuserrootpath.suffix=/
 wds.linkis.server.component.exclude.classes=org.apache.linkis.entranceclient.conf.ClientForEntranceSpringConfiguration,org.apache.linkis.entranceclient.conf.ClientSpringConfiguration,org.apache.linkis.entrance.conf.EntranceSpringConfiguration
+
 ##hive meta
 hive.meta.url=
 hive.meta.user=
 hive.meta.password=
+wds.linkis.metadata.hive.encode.enabled=false
```
Comment (peacewong) [inline linkis-dist/package/conf/linkis-ps-publicservice.properties:34]: This recommends separate pr and issue

File: linkis-dist/package/conf/linkis-ps-publicservice.properties [hunk 4]
```diff
@@ -13,25 +13,35 @@
...
+
 # associated with the logged-in user when querying metadata:default value is true
 #linkis.metadata.hive.permission.with-login-user-enabled
 #wds.linkis.jobhistory.undone.job.minimum.id=0
 #wds.linkis.jobhistory.undone.job.refreshtime.daily=00:15
+wds.linkis.cs.deserialize.replace_package_header.enable=false
```
Comment (peacewong) [inline linkis-dist/package/conf/linkis-ps-publicservice.properties:40]: This recommends separate pr and issue

File: linkis-dist/package/conf/linkis-ps-publicservice.properties [hunk 5]
```diff
@@ -13,25 +13,35 @@
...
 
 ##Spring
 spring.server.port=9105
 spring.spring.main.allow-bean-definition-overriding=true
+spring.spring.jackson.serialization.FAIL_ON_EMPTY_BEANS=false
```
Comment (peacewong) [inline linkis-dist/package/conf/linkis-ps-publicservice.properties:45]: This recommends separate pr and issue

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-client/src/main/scala/org/apache/linkis/instance/label/client/InstanceLabelClient.scala [hunk 6]
```diff
@@ -92,8 +92,8 @@ class InstanceLabelClient extends Logging {
           }
           resp.getInsList
         case o =>
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-instance-label/linkis-instance-label-client/src/main/scala/org/apache/linkis/instance/label/client/InstanceLabelClient.scala:94]: Maybe modifying to case _ is better

**Review by peacewong — APPROVED** [review]

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala (hunk 2 above)
Comment (peacewong) [inline linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala:66]: contextService is better

Comment (jackxu2011) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala:66]: the contextService's servicename is ps-cs,  this config is used to match the service name.

File: linkis-dist/package/conf/linkis-ps-publicservice.properties (hunk 5 above)
Comment (jackxu2011) [reply linkis-dist/package/conf/linkis-ps-publicservice.properties:45]: this config is merge form data-source-manager properties

File: linkis-dist/package/conf/linkis-ps-publicservice.properties (hunk 4 above)
Comment (jackxu2011) [reply linkis-dist/package/conf/linkis-ps-publicservice.properties:40]: this config is merged from linkis-ps-cs properties

File: linkis-dist/package/conf/linkis-ps-publicservice.properties (hunk 3 above)
Comment (jackxu2011) [reply linkis-dist/package/conf/linkis-ps-publicservice.properties:34]: this config is merged form linkis-ps-metadata-query

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-client/src/main/scala/org/apache/linkis/instance/label/client/InstanceLabelClient.scala (hunk 6 above)
Comment (jackxu2011) [reply linkis-public-enhancements/linkis-instance-label/linkis-instance-label-client/src/main/scala/org/apache/linkis/instance/label/client/InstanceLabelClient.scala:94]: the param o is used below

**Review by peacewong — APPROVED** [review]

**Comment by casionone** [comment]
```
public class MdmConfiguration {

  public static CommonVars<String> METADATA_SERVICE_APPLICATION =
      CommonVars.apply("wds.linkis.server.mdm.service.app.name", "linkis-ps-metadataquery");

  public static CommonVars<String> DATA_SOURCE_SERVICE_APPLICATION =
      CommonVars.apply("wds.linkis.server.dsm.app.name", "linkis-ps-data-source-manager");
}
```
need change to linkis-ps-publicserice

and the configuration of wds.linkis.gateway.conf.publicservice.list should be removed from linkis-dist/package/conf/linkis.properties

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala (hunk 2 above)
Comment (casionone) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/conf/RPCConfiguration.scala]: old version  use linkis-ps-metadatamanager, to ensure compatibility, it is recommended to add this `metadatamanager`

**Review by peacewong — APPROVED** [review]

**Pull request merged**